src/test/java/com/google/api/codegen/testdata/java/java_library.baseline

```diff
@@ -5125,196 +5128,232 @@ public class GrpcLibraryServiceStub extends LibraryServiceStub {
   private static final UnaryCallable<ListShelvesRequest, ListShelvesResponse> directListShelvesCallable =
       GrpcCallableFactory.createDirectCallable(
           io.grpc.MethodDescriptor.create(
               io.grpc.MethodDescriptor.MethodType.UNARY,
               "google.example.library.v1.LibraryService/ListShelves",
               io.grpc.protobuf.ProtoUtils.marshaller(ListShelvesRequest.getDefaultInstance()),
-              io.grpc.protobuf.ProtoUtils.marshaller(ListShelvesResponse.getDefaultInstance())));
+              io.grpc.protobuf.ProtoUtils.marshaller(ListShelvesResponse.getDefaultInstance())),
+          EmptyRequestParamsExtractor.<ListShelvesRequest>of());
   private static final UnaryCallable<DeleteShelfRequest, Empty> directDeleteShelfCallable =
       GrpcCallableFactory.createDirectCallable(
           io.grpc.MethodDescriptor.create(
               io.grpc.MethodDescriptor.MethodType.UNARY,
               "google.example.library.v1.LibraryService/DeleteShelf",
               io.grpc.protobuf.ProtoUtils.marshaller(DeleteShelfRequest.getDefaultInstance()),
-              io.grpc.protobuf.ProtoUtils.marshaller(Empty.getDefaultInstance())));
+              io.grpc.protobuf.ProtoUtils.marshaller(Empty.getDefaultInstance())),
+          EmptyRequestParamsExtractor.<DeleteShelfRequest>of());
   private static final UnaryCallable<MergeShelvesRequest, Shelf> directMergeShelvesCallable =
       GrpcCallableFactory.createDirectCallable(
           io.grpc.MethodDescriptor.create(
               io.grpc.MethodDescriptor.MethodType.UNARY,
               "google.example.library.v1.LibraryService/MergeShelves",
               io.grpc.protobuf.ProtoUtils.marshaller(MergeShelvesRequest.getDefaultInstance()),
-              io.grpc.protobuf.ProtoUtils.marshaller(Shelf.getDefaultInstance())));
+              io.grpc.protobuf.ProtoUtils.marshaller(Shelf.getDefaultInstance())),
+          EmptyRequestParamsExtractor.<MergeShelvesRequest>of());
   private static final UnaryCallable<CreateBookRequest, Book> directCreateBookCallable =
       GrpcCallableFactory.createDirectCallable(
           io.grpc.MethodDescriptor.create(
               io.grpc.MethodDescriptor.MethodType.UNARY,
               "google.example.library.v1.LibraryService/CreateBook",
               io.grpc.protobuf.ProtoUtils.marshaller(CreateBookRequest.getDefaultInstance()),
-              io.grpc.protobuf.ProtoUtils.marshaller(Book.getDefaultInstance())));
+              io.grpc.protobuf.ProtoUtils.marshaller(Book.getDefaultInstance())),
+          new RequestParamsExtractor<CreateBookRequest>() {
+            @Override
+            public Map<String, String> extract(CreateBookRequest request) {
+              ImmutableMap.Builder<String, String> params = ImmutableMap.builder();
+              params.put("name", String.valueOf(request.getName()));
+              params.put("book.read", String.valueOf(request.getBook().getRead()));
```

Comment:
I think it could be possible for a field like `request.getBook()` to be null.

Reply:
And in that case it will be "null" in the string. This is by design.

Reply:
I agree about failing fast, but I would prefer IllegalArgumentException over NullPointerException.

Reply:
I believe it is ok to allow NPE here. Otherwise it will significantly complicate snippets (the getters chain can be arbitrarily long, each component should be checked for non-null).
Also NPE are often considered ok and are thrown directly: `java.util.List.add(E)` and `com.google.common.base.Preconditions.checkNotNull()` are two basic examples, when `NPE` can be thrown directly.